### PR TITLE
Microsoft recommended azblob sdk has moved

### LINKED ISSUE #849: DecryptAndDecompressTar: lz4 decompress failed. Is archive encrypted?: DecompressLz4: lz4 write failed: unexpected EOF
Hi,

My DB server:
CentOS 7.7
PostgreSQL 12.5

I'm using WAL-G **v0.2.19** (Latest). Backups and WALs go to **S3 storage**. 

It's interesting that during restoration processes I'm getting some error messages:

```
NFO: 2021/01/15 11:10:36.969496 LATEST backup is: 'base_00000006000000000000000D
INFO: 2021/01/15 11:10:37.756309 Finished decompression of part_003.tar.lz4
INFO: 2021/01/15 11:10:37.756328 Finished extraction of part_003.tar.lz4
INFO: 2021/01/15 11:13:46.004024 Finished decompression of part_001.tar.lz4
ERROR: 2021/01/15 11:13:46.004063 part_001.tar.lz4 DecryptAndDecompressTar: lz4 decompress failed. Is archive encrypted?: DecompressLz4: lz4 write failed: unexpected EOF
INFO: 2021/01/15 11:13:46.004113 Finished extraction of part_001.tar.lz4
ERROR: 2021/01/15 11:13:46.004121 Extraction error in part_001.tar.lz4: extractOne: Interpret failed: Interpret: copy failed: unexpected EOF
INFO: 2021/01/15 11:16:04.996247 Finished decompression of part_001.tar.lz4
INFO: 2021/01/15 11:16:04.996269 Finished extraction of part_001.tar.lz4
INFO: 2021/01/15 11:16:05.003757 Finished decompression of pg_control.tar.lz4
INFO: 2021/01/15 11:16:05.003781 Finished extraction of pg_control.tar.lz4
INFO: 2021/01/15 11:16:05.003791 
Backup extraction complete.
```

Sometimes restore is OK, but sometimes it takes very long and finally crashed. 

I create backups in a regular way with backup-push (no encryption or other special elements).

The creation of backups and uploads to S3 storage is always normal and successful. Problems occur only with the restore process.

**What does it mean "DecryptAndDecompressTar: lz4 decompress failed. Is archive encrypted?: DecompressLz4: lz4 write failed: unexpected EOF"?** 

**Why does it happen?**

### PR DESCRIPTION
**Pull request by serprex**
https://github.com/Azure/azure-storage-blob-go/commit/b32b5e3579400739418747a71c4bffaf8a19c4a0

go get github.com/Azure/azure-sdk-for-go/sdk/storage/azblob
Replace import in pkg/storages/azure/folder.go with new azblob module
go mod tidy
& then fix everything

This PR is currently blocked on https://github.com/Azure/azure-sdk-for-go/pull/15958 unless Folder's `PutObject` is updated to take an `io.ReadSeekCloser` instead of `io.Reader` but `UploadStreamToBlockBlob` shouldn't be closing the object regardless so the PR should at least get it down to `io.ReadSeeker`. Will test once that PR settles

Also to consider: https://github.com/Azure/azure-sdk-for-go/blob/dfdc424c554cfb6e45a53e259c5c215bb4e1d3a6/sdk/storage/azblob/chunkwriting.go#L29 implies the default buffer size should be 8MB instead of 64MB

Fixes #849

### CODE REVIEW

**Comment by serprex** [comment]
The CI failures match issue I found trying to build with go 1.13; go 1.17 passes. I've opened an issue: https://github.com/Azure/azure-sdk-for-go/issues/15982

**Comment by x4m** [comment]
Why do we build on 1.13? Maybe it's time to upgrade?

File: pkg/storages/azure/folder.go [hunk 1]
```diff
@@ -99,40 +100,40 @@ func ConfigureFolder(prefix string, settings map[string]string) (storage.Folder,
 		tryTimeout = defaultTryTimeout
 	}
 
-	pipeLine := azblob.NewPipeline(
-		credential,
-		azblob.PipelineOptions{
-			Retry:      azblob.RetryOptions{TryTimeout: time.Duration(tryTimeout) * time.Minute},
```
Comment (serprex) [inline pkg/storages/azure/folder.go:105]: Need to figure out where try timeout goes now, also remove more code around here calculating the now unused `tryTimeout` variable

**Comment by serprex** [comment]
@x4m I was going to wait & see how high azure-sdk-for-go needed to bump their requirements, but considering #1116 I've updated the relevant Dockerfile to 1.17

**Comment by serprex** [comment]
@x4m Looks like my io.Reader PR isn't going through, considering I'm being pointed towards [`nopCloser`](https://github.com/Azure/azure-sdk-for-go/blob/dfdc424c554cfb6e45a53e259c5c215bb4e1d3a6/sdk/storage/azblob/internal/zc_shared.go#L27) suggesting that my hack is the de facto way to handle things

I've also changed from using `blobClient.Download` to `http.Get`, I'm torn on whether to apply `AZURE_TRY_TIMEOUT` to it, I'd prefer not to, but if you prefer it I'll update the code to use a timeout context. I've tested this on cases where we were hitting #849 in prod & the failures no longer appear

In short, I'm hoping to get this merged soon. I've tested uploading wal & pitr

**Review by x4m — APPROVED** [review]
Sorry, I did not understand what io.Reader PR you are referring. As to timeouts in Azure storage - I think you are the expert here. The code in this PR looks good to me, feel free to merge.

**Comment by serprex** [comment]
@x4m Ready to merge. The io.Reader PR I mentioned is https://github.com/Azure/azure-sdk-for-go/pull/15958

I did miss something beforehand: I forgot about account key auth. Implemented, mostly involved copying over signing code from sdk
